blivet/devices/storage.py

```diff
@@ -755,10 +755,7 @@ def _update_netdev_mount_option(self):
         is_netdev = any(isinstance(a, NetworkStorageDevice)
                         for a in self.ancestors)
         has_netdev_option = netdev_option in option_list
-        if not is_netdev and has_netdev_option:
-            option_list.remove(netdev_option)
-            self._format.options = ",".join(option_list)
-        elif is_netdev and not has_netdev_option:
+        if is_netdev and not has_netdev_option:
```

Comment:
The block that can remove the option is so that, if the parent devices were changed such that network-connected disks were switched out for locally-connected disks, we would do the right thing in updating the options. Obviously we need to be able to honor this explicitly-passed option somehow, but I don't think we can remove that block unconditionally.

Reply:
I've been thinking about this and i wasn't able to come up with situation where the parents would change after creating the format instance. Anyway, I've changed the code to only remove the option if it wasn't originally specified by the user and I've also added a test to make sure the option is removed when added by us.

Reply:
I don't know whether anyone uses this workflow, but it is possible to change the candidate disks and re-run `do_partitioning`, which could lead to the _netdev option being added or removed.